fix(linter): allow escaping at the start of a regex group (#5177)
biomejs/biome: .changeset/naming-conventions-regex-allow-char-escaping-at-start-of-regex-groups.md

---
"@biomejs/biome": patch
---

The lint rules `useNamingConvention` and `useFilenamingConvention` now accepts character escapes at the start of a regex group.

The rules `useNamingConvention` and `useFilenamingConvention` provides some options that allow matching names against a regular expression.

> Previously, escaped character at the start of a regex group reported an error.

Reviewer: ```suggestion
Previously, an escaped character at the start of a regex group reported an error.
```

They are now accepted.

> For example, the following configuration used to emit en error.
> It no longer raised an error.

Reviewer: ```suggestion
For example, the following configuration is now valid doesn't emit an error anymore.
```
